napari/_vispy/layers/vectors.py

```diff
@@ -377,25 +381,20 @@ def generate_meshes_arrow_2D(vectors, width, length, p):
 
     vertices = vertices + width * offsets / 2
 
-    triangles = np.array(
-        [
-            [7 * i / 3, 7 * i / 3 + 1, 7 * i / 3 + 2]
-            if i % 3 == 0
-            else [
-                7 * (i - 1) / 3 + 1,
-                7 * (i - 1) / 3 + 2,
-                7 * (i - 1) / 3 + 3,
-            ]
-            if i % 3 == 1
-            else [
-                7 * (i - 2) / 3 + 4,
-                7 * (i - 2) / 3 + 5,
-                7 * (i - 2) / 3 + 6,
-            ]
-            for i in range(3 * nvectors)
-        ]
-    ).astype(np.uint32)
-
-    # np.tile([[0,1,2],[1,2,3],[4,5,6]], (nvectors, 1)) + np.repeat(7*np.arange(nvectors), 3)[:,None]
+    # Generate triangles in two steps:
+    # 1. Repeat the vertices pattern
+    # [[0,1,2],
+    #  [1,2,3]
+    #  [4,5,6]]
+    # as described in the docstring
+    vertices_pattern = np.tile(
+        [[0, 1, 2], [1, 2, 3], [4, 5, 6]], (nvectors, 1)
+    )
+    # 2. Add an offset to differentiate between vectors
+    triangles = (
+        vertices_pattern + np.repeat(4 * np.arange(nvectors), 2)[:, None]
+    )
+
```

Comment:
oh wow, I didn't even know about slicing with `None`! Pretty cool :)

Reply:
Some people consider this bad practice, as `np.newaxis` should be used instead of `None`... Don't hesitate if you would  like me to change it. :)

Reply:
I guess it's probably wise since it's more well known :)